### PR TITLE
Two media library issues

### PR DESCRIPTION
**Pull request by nbradbury**
This PR resolves two of the issues listed in #5337:

* Drops the ability to create a gallery from the media browser. This feature may be redesigned and reintroduced down the road, but for now it's not very friendly so we've decided to remove it. Note that the user can still select multiple photos, but rather than being inserted into a new post as a gallery shortcode they're inserted as separate photos.

* Drops the custom date filter from the media browser. This feature didn't work in FluxC and doesn't exist in Calypso, so it was decided that it should be removed.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridFragment.java [hunk 1]
```diff
@@ -630,20 +626,6 @@ public void onClick(DialogInterface dialog, int which) {
...
-            return;
-        }
-
-        ArrayList<Long> remoteMediaIds = new ArrayList<>();
-        for (int localMediaId : mGridAdapter.getSelectedItems()) {
-            MediaModel mediaModel = mMediaStore.getMediaWithLocalId(localMediaId);
-            remoteMediaIds.add(mediaModel.getMediaId());
-        }
-
-        ActivityLauncher.newGalleryPost(getActivity(), mSite, remoteMediaIds);
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridFragment.java]: This leaves `ActivityLauncher.newGalleryPost()` unused - I think we should probably clean that up too.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 2]
```diff
@@ -1093,7 +1093,7 @@ public int getCount() {
     public static final String NEW_MEDIA_GALLERY = "NEW_MEDIA_GALLERY";
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: If we drop `ActivityLauncher.newGalleryPost()`, we can also drop `NEW_MEDIA_GALLERY`, `NEW_MEDIA_GALLERY_EXTRA_IDS`, and `EditPostActivity.prepareMediaGallery()`.

**Comment by aforcier** [comment]
Looks great @nbradbury! Just had a couple of really minor cleanup comments.

**Comment by nbradbury** [comment]
Thanks for the review @aforcier! Good call on the cleanup - addressed in 2f1fe0c.

**Comment by aforcier** [comment]
:shipit: - just waiting for Travis.

**Pull request merged**